src/main/java/com/artipie/npm/proxy/NpmRemoteImpl.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 artipie
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.npm.proxy;
+
+import com.artipie.asto.fs.RxFile;
+import com.artipie.npm.proxy.json.CachedContent;
+import com.artipie.npm.proxy.model.NpmAsset;
+import com.artipie.npm.proxy.model.NpmPackage;
+import com.jcabi.log.Logger;
+import io.reactivex.Maybe;
+import io.vertx.core.file.OpenOptions;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.web.client.WebClient;
+import io.vertx.reactivex.ext.web.codec.BodyCodec;
+import java.nio.file.Path;
+
+/**
+ * NPM Remote client implementation.
+ * @since 0.1
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class NpmRemoteImpl implements NpmRemote {
```

Comment:
@Vatavuk Ok, I'll rename to fit naming conventions, but we will not have multiple implementations. And anyway it will be HTTP-based ) The only reason to extract this interface is the fact, that Mockito can not mocks final classes (it really can with new extra tool, but it uses bytecode generator inside and I try to avoid this when it possible). The same for NpmStorage